### PR TITLE
fix(annotations): throw error when all adds or updates fail

### PR DESCRIPTION
**Pull request by tomwayson**
affects: @esri/hub-annotations

ISSUES CLOSED: #430

### CODE REVIEW

**Comment by tomwayson** [comment]
@lisastaehli once this is merged the add and update annotation functions will throw an error if all results failed.

File: packages/annotations/src/add.ts [hunk 1]
```diff
@@ -43,14 +44,19 @@ export interface IAddAnnotationsOptions extends IAddFeaturesOptions {
...
     requestOptions.features.forEach(anno => enrichAnnotation(anno));
   }
 
-  return addFeatures(requestOptions);
+  return addFeatures(requestOptions).then(response => {
+    return {
+      ...response,
+      addResults: checkResults(response.addResults)
+    };
+  });
```
Comment (mikeringrose) [inline packages/annotations/src/add.ts:59]: Does this align with the return type?
```
Promise<{
  addResults?: Array<IEditFeatureErrorResponse | IEditFeatureResult>;
}>
```

Comment (tomwayson) [reply packages/annotations/src/add.ts:59]: yep, in fact TS complained when I made a typo and it wasn't aligned.

Comment (mikeringrose) [reply packages/annotations/src/add.ts:59]: even though it mentions nothing of the destructured response object. strange.

Comment (mikeringrose) [reply packages/annotations/src/add.ts:59]: never mind. i didn't see the destructuring at first.

Comment (mikeringrose) [reply packages/annotations/src/add.ts:59]: the destructuring doesn't seem necessary since your always overriding, but it's already merged.

Comment (tomwayson) [reply packages/annotations/src/add.ts:59]: I did the destructuring b/c thought there was additional properties in the response. looking at the [API docs](https://developers.arcgis.com/rest/services-reference/add-features.htm) there may not be.

**Pull request merged**